### PR TITLE
isort convolve

### PR DESCRIPTION
**Pull request by nstarman**
Signed-off-by: Nathaniel Starkman (@nstarman) <nstarkman@protonmail.com>

Running ``isort`` on ``astropy/convolution``  and adding ``__all__`` to ``__init__.py``.
The exact command was ``>>> isort astropy/convolution/ -l 100``

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/convolution/__init__.py [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
-# Deprecated kernels that are not defined in __all__
-from .kernels import MexicanHat1DKernel, MexicanHat2DKernel
+__all__ = (
+    ["convolve", "convolve_fft", "convolve_models", "convolve_models_fft",
+     "interpolate_replace_nans"]
+    + core.__all__
+    + kernels.__all__
+    + ["MexicanHat1DKernel", "MexicanHat2DKernel"]
+    + utils.__all__
+)
```
Comment (astrofrog) [inline astropy/convolution/__init__.py:19]: I don't think we should do this - in general we have followed the pattern of not having __all__ defined in top level unit files, and if we are going to programmatically populate it like this it really serves no purpose?

**Review by keflavich — APPROVED** [review]
lgtm; this is all just formatting changes

File: astropy/convolution/__init__.py (hunk 1 above)
Comment (keflavich) [reply astropy/convolution/__init__.py:19]: I changed my review away from 'approved' b/c of this - I don't understand the broader implications of the proposed change.

Comment (nstarman) [reply astropy/convolution/__init__.py:19]: I will remove it.
I personally like having ``__all__`` defined at every level to never break expectations — ``__all__`` is always defined.

Comment (nstarman) [reply astropy/convolution/__init__.py]: Removed.

**Pull request merged**